test/rubygems/test_require.rb

```diff
@@ -247,6 +247,32 @@ def test_unable_to_find_good_unresolved_version
     assert_equal "unable to find a version of 'b' to activate", e.message
   end
 
+  def test_require_works_after_cleanup
+    a1 = new_default_spec "a", "1.0", nil, "a/b.rb"
+    b1 = new_default_spec "b", "1.0", nil, "b/c.rb"
+    b2 = new_default_spec "b", "2.0", nil, "b/d.rb"
+
+    install_default_gems a1
+    install_default_gems b1
+    install_default_gems b2
+
+    # Load default ruby gems fresh as if we've just started a ruby script.
+    Gem::Specification.reset
+    require 'rubygems'
+    Gem::Specification.stubs
+
+    # Remove an old default gem version directly from disk as if someone ran
+    # gem cleanup.
+    FileUtils.rm_rf(File.join @default_dir, "#{b1.full_name}")
+    FileUtils.rm_rf(File.join @default_spec_dir, "#{b1.full_name}.gemspec")
+
+    # Require gems that have not been removed.
+    assert_require 'a/b'
```

Comment:
Run against the current master code, this raises the "undefined method 'platform'" exception.  With the changes in this PR, it passes (no exception is raised).